src/main/perl/Download.pm

```diff
@@ -0,0 +1,142 @@
+# ${license-info}
+# ${developer-info}
+# ${author-info}
+# ${build-info}
+
+package CAF::Download;
+
+use strict;
+use warnings;
+
+# For re-export only
+use CAF::Download::URL qw(set_url_defaults);
+use parent qw(CAF::ObjectText Exporter CAF::Download::URL CAF::Download::Retrieve);
+
+use Readonly;
+use LC::Exception qw (SUCCESS);
+
+our @EXPORT_OK = qw(set_url_defaults);
+
+# TODO: dependencies on curl and kinit
+
+Readonly::Hash my %DOWNLOAD_METHODS => {
+    http => [qw(lwp curl)], # try https, if not, try http
```

Comment:
It's not obvious how that is related to lwp & curl. Falling back to an insecure method by default is bad but having it configurable is good.